src/multiline/flb_ml.c

```diff
@@ -1443,6 +1443,7 @@ int flb_ml_flush_stream_group(struct flb_ml_parser *ml_parser,
         }
         msgpack_unpacked_destroy(&result);
         group->mp_sbuf.size = 0;
+        group->mp_md_sbuf.size = 0;
```

Comment:
while I don't remember all the details of the implementation and it seems metadata buffer is newer, why needs to be set to zero at this stage ?

also, I don't see other part in the code where this is set to zero, maybe we are missing another fix around ?

just comments mostly.